index.js

```diff
@@ -48,7 +58,9 @@ const createLogs = (dataForLogging) => {
     }
   };
 
-  lineToInsert = isNewLine() + JSON.stringify(dataForLogging);
+  const isThereAnError = errorsWhileGettingData ? errorsWhileGettingData : dataForLogging;
```

Comment:
I don't think that `createLogs` function should know anything about external context (if any error happened before or not) — it can simply write to the file whatever comes as `dataForLogging`. So we'll just call `createLogs(error)` if we want to write error to a log.

It seems you don't need these changes if you apply the suggestion above (moving `try/catch` block from `fetchData` to `collectDataOnceInAMinute`).